modules/renter/contractor/update.go

```diff
@@ -53,16 +53,16 @@ func (c *Contractor) ProcessConsensusChange(cc modules.ConsensusChange) {
 			}
 			defer c.editLock.Unlock()
 
-			// renew any contracts that have entered the renew window
+			// renew any (online) contracts that have entered the renew window
 			err := c.managedRenewContracts()
 			if err != nil {
 				c.log.Debugln("WARN: failed to renew contracts after processing a consensus chage:", err)
 			}
 
-			// if we don't have enough contracts, form new ones
+			// if we don't have enough (online) contracts, form new ones
 			c.mu.RLock()
 			a := c.allowance
-			remaining := int(a.Hosts) - len(c.contracts)
+			remaining := int(a.Hosts) - len(c.onlineContracts())
```

Comment:
I've never heard of this rule. Our only official convention regarding unexported function locking is:

>Non-exported functions should not do any locking, unless they have a special prefix to the name (explained below). The responsibility for thread-safety comes from the exported functions which call the non-exported functions.

Nothing in there about it needing to be a write lock.